src/login/loginctl.c

```diff
@@ -497,101 +489,164 @@ static int print_session_status_info(sd_bus *bus, const char *path, bool *new_li
+                _cleanup_free_ char *name = NULL;
 
-                printf("\t  Leader: " PID_FMT, i.leader);
+                (void) get_process_comm(i.leader, &name);
 
-                (void) get_process_comm(i.leader, &t);
-                if (t)
-                        printf(" (%s)", t);
+                r = table_add_cell(table, NULL, TABLE_FIELD, "Leader");
+                if (r < 0)
+                        return table_log_add_error(r);
 
-                printf("\n");
+                r = table_add_cell_stringf(table, NULL, PID_FMT "%s%s%s",
+                                           i.leader,
+                                           !isempty(name) ? " (" : "",
+                                           strempty(name),
+                                           !isempty(name) ? ")" : "");
+                if (r < 0)
+                        return table_log_add_error(r);
         }
 
-        if (!isempty(i.seat)) {
-                printf("\t    Seat: %s", i.seat);
-
-                if (i.vtnr > 0)
-                        printf("; vc%u", i.vtnr);
+        r = table_add_cell(table, NULL, TABLE_FIELD, "Seat");
+        if (r < 0)
+                return table_log_add_error(r);
 
-                printf("\n");
-        }
+        if (i.vtnr > 0)
+                r = table_add_cell_stringf(table, NULL, "%s; vc%u", i.seat, i.vtnr);
+        else
+                r = table_add_cell(table, NULL, TABLE_STRING, i.seat);
+        if (r < 0)
+                return table_log_add_error(r);
 
         if (i.tty)
-                printf("\t     TTY: %s\n", i.tty);
+                r = table_add_many(table,
+                                   TABLE_FIELD, "TTY",
+                                   TABLE_STRING, i.tty);
         else if (i.display)
-                printf("\t Display: %s\n", i.display);
+                r = table_add_many(table,
+                                   TABLE_FIELD, "Display",
+                                   TABLE_STRING, i.display);
+        else
+                r = 0;
+        if (r < 0)
+                return table_log_add_error(r);
+
+        r = table_add_cell(table, NULL, TABLE_FIELD, "Remote");
+        if (r < 0)
+                return table_log_add_error(r);
 
         if (i.remote_host && i.remote_user)
-                printf("\t  Remote: %s@%s\n", i.remote_user, i.remote_host);
+                r = table_add_cell_stringf(table, NULL, "%s@%s", i.remote_user, i.remote_host);
         else if (i.remote_host)
-                printf("\t  Remote: %s\n", i.remote_host);
+                r = table_add_cell(table, NULL, TABLE_STRING, i.remote_host);
         else if (i.remote_user)
-                printf("\t  Remote: user %s\n", i.remote_user);
-        else if (i.remote)
-                printf("\t  Remote: Yes\n");
+                r = table_add_cell_stringf(table, NULL, "user %s", i.remote_user);
+        else
+                r = table_add_cell(table, NULL, TABLE_BOOLEAN, &i.remote);
```

Comment:
Here, a pointer to boolean (with `false`) should be passed.

Reply:
Hmm, I don't really follow? &SessionStatusInfo.remote is a pointer to boolean. What do you mean by "with false"?

Reply:
Hoops, sorry for the noise. Please forget.